tests/shouldwork/Netlist/NoDeDup.hs

```diff
@@ -0,0 +1,74 @@
+{-# LANGUAGE ViewPatterns #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module NoDeDup where
+
+import Prelude as P
+
+import Clash.Magic
+import Clash.Prelude
+import Clash.Netlist.Types
+
+import Test.Tasty.Clash
+import Test.Tasty.Clash.NetlistTest
+
+data ABCD = A | B | C | D
+
+twice :: Int -> Int
+twice a = 2 * a
+{-# NOINLINE twice #-}
+
+-- | 'f' should have one call to twice in the netlist
+f :: Int -> ABCD -> Int
+f n abcd =
+  case abcd of
+    A -> twice (6 + n)
+    B -> 5 + n
+    C -> n - 3
+    D -> twice (3 + n)
+{-# NOINLINE f #-}
+
+-- | 'g' should have two calls to twice in the netlist
+g :: Int -> ABCD -> Int
+g n abcd =
+  case abcd of
+    A -> noDeDup (twice (6 + n))
+    B -> 5 + n
+    C -> n - 3
+    D -> noDeDup (twice (3 + n))
+{-# NOINLINE g #-}
+
+topEntity :: Int -> ABCD -> Int
+topEntity n abcd = (f n abcd) - (g n abcd)
+
+testPath :: FilePath
+testPath = "tests/shouldwork/Netlist/NoDeDup.hs"
+
+isTwiceInst (InstDecl Entity Nothing "twice" _ _ _) = True
+isTwiceInst _ = False
+
+assertAssignsInOut :: Component -> IO ()
+assertAssignsInOut (Component nm inps outs ds) =
```

Comment:
`assertAssignsInOut` was picked because it was more specific to the example netlist test I defined. I would probably rename this to something more meaningful for this test (like `assertNumTwiceInsts`)

Reply:
Woops, yes! I just mindlessly copied it.

Reply:
Done, thanks for catching that.